app/router.js

```diff
@@ -1,9 +1,33 @@
 import EmberRouter from '@ember/routing/router';
+
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { scheduleOnce } from '@ember/runloop';
+
 import config from './config/environment';
 
 const Router = EmberRouter.extend({
   location: config.locationType,
-  rootURL: config.rootURL
+  rootURL: config.rootURL,
+
+  metrics: service(),
+
+  didTransition() {
+    this._super(...arguments);
+    this._trackPage();
+  },
+
+  _trackPage() {
+    scheduleOnce('afterRender', this, () => {
+      const page = this.get('url');
+      const title = this.getWithDefault('currentRouteName', 'unknown');
+
+      // this is constant for this app and is only used to identify page views in the GA dashboard
+      const hostname = 'guides.emberjs.com';
+
+      get(this, 'metrics').trackPage({ page, title, hostname });
```

Comment:
I think we should settle on a standard nomenclature as a team. I don’t like this destructuring approach myself as I think it’s likely to be less code-modable ...